frontend/src/app/PrimeErrorBoundary.test.tsx

```diff
@@ -0,0 +1,41 @@
+import { render } from "@testing-library/react";
+import { MemoryRouter } from "react-router-dom";
+import { Component } from "react";
+
+import PrimeErrorBoundary from "./PrimeErrorBoundary";
+
+jest.mock("./AppInsights", () => {
+  return {
+    appInsights: { trackException: jest.fn() },
+  };
+});
+
+class ComponentOfDeath extends Component<any, any> {
+  public componentDidMount() {
+    throw new Error("a really terrible uncaught exception!");
+  }
+  public render() {
+    return <h1>Component of doooom</h1>;
+  }
+}
+
+describe("PrimeErrorBoundary", () => {
+  beforeEach(() => {
+    jest.spyOn(console, "error").mockImplementation(() => {});
+  });
+
+  afterEach(() => {
+    jest.spyOn(console, "error").mockRestore();
+  });
+
+  it("renders the error page on component error", () => {
+    const { container } = render(
+      <MemoryRouter>
+        <PrimeErrorBoundary>
+          <ComponentOfDeath></ComponentOfDeath>
+        </PrimeErrorBoundary>
+      </MemoryRouter>
+    );
+    expect(container).toHaveTextContent("Something went wrong :(");
```

Comment:
Great test!!

Reply:
Thanks! And thanks for the resources you sent me, they helped a lot.